Source/dom/style/Style.js

```diff
@@ -181,3 +199,56 @@ Style.define('innerShadows', {
     this._object.setInnerShadows(objects)
   },
 })
+
+Style.define('sharedStyleId', {
+  get() {
+    if (this._object.sharedObjectID()) {
+      return String(this._object.sharedObjectID())
+    }
+    return null
+  },
+  set(newId) {
+    const documentData = this._object.documentData()
+    const container = documentData.sharedObjectContainerOfType(
+      this._object.type()
+    )
+
+    if (!newId) {
+      const currentSharedId = this._object.sharedObjectID()
+      if (currentSharedId) {
+        const currentSharedObject = container.sharedStyleWithID(currentSharedId)
+        currentSharedObject.unregisterInstance(this._object)
+      }
+    } else {
+      const sharedStyle = container.sharedStyleWithID(newId)
+      if (!sharedStyle) {
+        throw new Error('Seems like this shared style does not exists')
+      }
+      container.registerInstance_withSharedStyle(this._object, sharedStyle)
+    }
```

Comment:
The styles are still in the shared containers but they aren't nice, and the plan is to gradually move away from them until we get to the point where we can just rip them out. A better way to do this would be to use the `sharedStyle` property defined in `MSLayer+SharedStyles.h`.